src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -639,6 +639,9 @@ TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extent(
   assert(extent->is_valid() && !extent->is_initial_pending());
   if (extent->is_dirty()) {
     if (epm->can_inplace_rewrite(t, extent)) {
+      // FIXME: is_dirty() is true for mutation pending extents
+      // which shouldn't do inplace rewrite because a pending transaction
+      // may fail.
```

Comment:
Seems to be a gap as described @myoungwon

Reply:
Do you think `if (extent->state == CachedExtent::extent_state_t::DIRTY)`would be more accurate instead of `is_dirty`? Did I understand your concern correctly?

Reply:
I think if the extent is (existing-)mutation-pending, and if its prior-instance is DIRTY, inplace-rewrite should only apply to the shared "DIRTY" instance, and make sure the "mutation-pending" part keeps transactional until committed (i.e. the pending part should not apply inplace-rewrite).

However, I think the above can can be simplified to exclude all the mutation-pending extents from the inplace-rewrite path.

BTW, if you think the comments make sense, can you approve this?

Reply:
I see.